config/v1/types_infrastructure.go

```diff
@@ -531,9 +531,296 @@ type OvirtPlatformStatus struct {
+	// vm guests are located. This field will
+	// be used by the Cloud Controller Manager.
+	// Each datacenter listed here should be used within
+	// a topology.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Datacenters []string `json:"datacenters"`
+}
+
+// VSpherePlatformPlacementConstraint is the context information for VM placements within a failure domain
+type VSpherePlatformPlacementConstraint struct {
+	// resourcePool is the absolute path of the resource pool where virtual machines will be
+	// created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:Pattern=`^/.*?/host/.*?/Resources.*`
+	ResourcePool string `json:"resourcePool,omitempty"`
+
+	// folder is the name or inventory path of the folder in which the
+	// virtual machine is created/located.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	Folder string `json:"folder,omitempty"`
+}
+
+// VSpherePlatformDeploymentSuitable
+// +kubebuilder:validation:Enum=Allowed;NotAllowed
+type VSpherePlatformDeploymentSuitable string
+
+const (
+	// Allowed indicates that the Deployment Zone is suitable for
+	// control plane nodes.
+	Allowed VSpherePlatformDeploymentSuitable = "Allowed"
+
+	// NotAllowed indicates that the Deployment Zone is not suitable for
+	// control plane nodes.
+	NotAllowed VSpherePlatformDeploymentSuitable = "NotAllowed"
+)
+
+// VSpherePlatformDeploymentZone holds the association between a
+// vCenter, failure domain and the virtual machine placementConstraints
+type VSpherePlatformDeploymentZone struct {
+	// name defines the VSpherePlatformDeploymentZoneSpec name.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// server is the fully-qualified domain name or the IP address of the vCenter server.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=255
+	Server string `json:"server"`
+
+	// failureDomain is the name of the VSphereFailureDomain used for this VSphereDeploymentZone
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	FailureDomain string `json:"failureDomain"`
+
+	// ControlPlane determines if this failure domain is suitable for use by control plane machines.
+	// There is three valid options: unset, Allowed and NotAllowed.
+	// +kubebuilder:validation:Optional
+	ControlPlane VSpherePlatformDeploymentSuitable `json:"controlPlane,omitempty"`
+
+	// PlacementConstraint encapsulates the placement constraints
+	// used within this deployment zone.
+	// +kubebuilder:validation:Optional
```

Comment:
I'm gonna stop doing this but please find and replace on these
```suggestion
	// +optional
```